src/app/ReadClient.cpp

```diff
@@ -435,6 +437,11 @@ CHIP_ERROR ReadClient::OnMessageReceived(Messaging::ExchangeContext * apExchange
     }
 
 exit:
+    if (err != CHIP_NO_ERROR)
+    {
+        StatusResponse::Send(Status::InvalidAction, apExchangeContext, false /*aExpectResponse*/);
```

Comment:
This is sending the wrong status if we get here with a ReportData that has an incorrect subscription id.  Followup for this is best, I think, to keep scope down, esp. since https://github.com/project-chip/connectedhomeip/issues/19130 is tracking it...  But might be worth adding a TODO comment that on subscription id mismatch this should send InvalidSubscription, not InvalidAction.

Reply:
let me add TODO in case.